### DISCUSSION TITLE
Restrict <Drag> to single axis

### DISCUSSION THREAD

**Discussion by valtism** [airbnb/visx#1365] [Q&A] [answered]
Hi there 👋

Wondering if there is a way to restrict the `<Drag>` component to a single axis. I am trying to build a tool like this [Startup Growth Calculator](http://growth.tlb.org) which has handles that can be dragged to adjust lines on the chart. 

Is something like this possible using `<Drag>`? If not I'm sure I can implement it with a custom implementation using event listeners.

**Comment by williaster** [comment, marked as answer]
Hey @valtism this is a great suggestion! Although it's not currently supported I think this would be useful and would probably not be too difficult to add to `Drag`. Happy to talk about approaches and/or review a PR if someone is interested in adding it! 🔥

**Comment by valtism** [comment]
@williaster Starting some work towards this here: https://github.com/airbnb/visx/pull/1368
